contrib/generate_precompile.jl

```diff
@@ -245,15 +245,16 @@ function generate_precompile_statements()
     sysimg = Base.unsafe_string(Base.JLOptions().image_file)
 
     # Extract the precompile statements from the precompile file
-    statements = Set{String}()
+    statements_step1 = Channel{String}(Inf)
+    statements_step2 = Channel{String}(Inf)
 
     # From hardcoded statements
     for statement in split(hardcoded_precompile_statements::String, '\n')
-        push!(statements, statement)
+        push!(statements_step1, statement)
     end
 
     # Collect statements from running the script
-    mktempdir() do prec_path
+    @async mktempdir() do prec_path
```

Comment:
These should have `errormonitor` on them so that their failure is loud

Reply:
Thank you, should be loud now.

Reply:
@vchuravy  `errormonitor` causes CI to fail for some reason. Is the current approach with `fetch(step1) == :ok || throw("Step 1 of collecting precompiles failed.")` sufficient?